iceoryx_utils/include/iceoryx_utils/internal/concurrent/lockfree_queue/resizeable_lockfree_queue.inl

```diff
@@ -0,0 +1,233 @@
+{
+    setCapacity(initialCapacity);
+}
+
+template <typename ElementType, uint64_t MaxCapacity>
+constexpr uint64_t ResizeableLockFreeQueue<ElementType, MaxCapacity>::maxCapacity() noexcept
+{
+    return MAX_CAPACITY;
+}
+
+template <typename ElementType, uint64_t MaxCapacity>
+uint64_t ResizeableLockFreeQueue<ElementType, MaxCapacity>::capacity() const noexcept
+{
+    return m_capacity.load(std::memory_order_relaxed);
+}
+
+
+template <typename ElementType, uint64_t MaxCapacity>
+template <typename ContainerType, typename>
+bool ResizeableLockFreeQueue<ElementType, MaxCapacity>::setCapacity(uint64_t newCapacity,
+                                                                    ContainerType& removedElements) noexcept
+{
+    //@todo: preferably the signature should be void(ElementType&) but this leads to a compile error
+    //       reason for change is to reflect the fact that the elements may be changed
+    //       by the removeHandler (main use case is move)
+    auto removeHandler = [&](const ElementType& value) { removedElements.emplace_back(std::move(value)); };
+    return setCapacity(newCapacity, removeHandler);
+}
+
+template <typename ElementType, uint64_t MaxCapacity>
+bool ResizeableLockFreeQueue<ElementType, MaxCapacity>::setCapacity(uint64_t newCapacity) noexcept
+{
+    auto removeHandler = [](const ElementType&) {};
+    return setCapacity(newCapacity, removeHandler);
+}
+
+template <typename ElementType, uint64_t MaxCapacity>
+template <typename Function, typename>
+bool ResizeableLockFreeQueue<ElementType, MaxCapacity>::setCapacity(uint64_t newCapacity,
+                                                                    Function&& removeHandler) noexcept
+{
+    if (newCapacity > MAX_CAPACITY)
+    {
+        return false;
+    }
+
+    if (m_resizeInProgress.test_and_set(std::memory_order_acquire))
+    {
+        // at most one resize can be in progress at any time
+        return false;
+    }
+
+    auto cap = capacity();
+
+    while (cap != newCapacity)
+    {
+        if (cap < newCapacity)
+        {
+            auto toIncrease = newCapacity - cap;
+            increaseCapacity(toIncrease); // return value does not matter, we check the capacity later
+        }
+        else
+        {
+            auto toDecrease = cap - newCapacity;
+            decreaseCapacity(toDecrease, removeHandler); // return value does not matter, we check the capacity later
+        }
+
+        cap = capacity();
+    }
+
+    // sync everything related to capacity change, e.g. the new capacity stored in m_capacity
+    m_resizeInProgress.clear(std::memory_order_release);
+    return true;
+}
+
+template <typename ElementType, uint64_t MaxCapacity>
+uint64_t ResizeableLockFreeQueue<ElementType, MaxCapacity>::increaseCapacity(uint64_t toIncrease) noexcept
+{
+    // we can be sure this is not called concurrently due to the m_resizeInProgress flag
+    //(this must be ensured as the vector is modified)
+    uint64_t increased = 0;
```

Comment:
Please write 0U

Reply:
OK, but I dislike how some AUTOSAR rules lead to worse code for no good reason.